CrazyCanvas/Source/ECS/Systems/Player/WeaponSystem.cpp

```diff
@@ -470,7 +481,7 @@ void WeaponSystem::TryFire(
 		}
 
 		// Fire the gun
-		weaponComponent.CurrentAmmunition--;
+		weaponComponent.WeaponTypeAmmo[ammoType].first--;
```

Comment:
Det är ganska kostsamt att hitta en entry i en map, du letar efter [ammoType] ovan, på rad 473. Bättre att spara resultat av den "sökningen" (som referens!) och sedan återanvända den igen.

Reply:
Alright, thanks you for pointing that out! =)